storybookjs/storybook
# Feature: Extend setAddon API

### The current API:

```js
setAddon({ addonA(), addonB() }) 
```
It adds addons to `storiesOf()` object and they accessible as methods:

```js
storiesOf(...)
  .addonA()
  .addonB()
```
### Why we need new API

Sometimes we want to implement some features or some behavior to `storiesOf()` to have it by default. This would open up a plenty new directions in the addons creation!

### What I did

Just allow passing into `setAddon` functions matching `/^_init/` and make them executed right after `storiesOf()` is created.
This is a backward compatible change, addons with normal names still work the same.

### New API

```js
setAddon({ addonA(), addonB(), _initSmth() }) 
```
when you create:

```js
storiesOf(...)
  .addonA()
  .addonB()
```

`_initSmth()` will not be accessible as a method of `storiesOf()` but it will be already executed

### How to test

run cra-kitchen-sink
check out the Toc first story of each storyKind (it just an example of how it works)
check that other addons work as usual
